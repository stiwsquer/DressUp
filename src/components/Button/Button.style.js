import styled from "styled-components";

export const StyledButton = styled.button`
  background: #202020;
  color: white;
  border: none;
  outline: none;
  padding: 1rem 3rem;
  letter-spacing: 0.15rem;
  font-weight: 900;
  cursor: pointer;
  &:hover {
    opacity: 0.7;
  }
`;
