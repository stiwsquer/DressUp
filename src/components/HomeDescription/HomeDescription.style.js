import styled from "styled-components";
import { flex } from "../../GlobalStyles";

export const HomeDescriptionSection = styled.section`
  min-height: 40vh;
  width: 95%;
  margin: auto;
  text-align: center;
  line-height: 3rem;

  ${flex({ direction: "column", justify: "center", align: "center" })}

  h2 {
    font-size: 2.5rem;
    margin: 4rem;
  }
  p {
    font-size: 1.5rem;
  }
  button {
    margin: 4rem;
  }
`;
