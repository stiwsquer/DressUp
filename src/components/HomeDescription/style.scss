@import "../../styles/global.scss";

.home-description {
  min-height: 40vh;
  width: 95%;
  margin: auto;
  text-align: center;
  line-height: 3rem;
  @include flex(column, center, center);
  h2 {
    font-size: $header3;
    margin: 4rem;
  }
  p {
    font-size: 1.5rem;
  }
  button {
    margin: 4rem;
  }
}
