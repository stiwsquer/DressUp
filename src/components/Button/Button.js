import React from 'react';
import { StyledButton } from './Button.style';

export default React.memo(({ children, type, className, onClick }) => (
  <StyledButton onClick={onClick} className={className} type={type}>
    {children}
  </StyledButton>
));
