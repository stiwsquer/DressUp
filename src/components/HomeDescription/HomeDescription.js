import React from "react";
import Button from "../Button/Button";
import { HomeDescriptionSection } from "./HomeDescription.style";
export default React.memo(function HomeDescription() {
  return (
    <HomeDescriptionSection>
      <h2>
        Trendy Clothing & Accessories at Dress Up - An Online Dress Boutique
      </h2>
      <p>
        DRESS UP IS A WOMEN&apos;S CLOTHING BOUTIQUE CARRYING A VARIETY OF FITS,
        STYLES, AND PATTERNS FOR WOMEN OF ALL AGES, SHAPES, AND SIZES. WE OFFER
        A DIVERSE COLLECTION OF NEW PRODUCTS EVERY SINGLE WEEK, INCLUDING YOUR
        FAVORITE CLOTHING ITEMS: CUTE ROMPERS, MAXI DRESSES, KNIT SWEATERS,
        OVERSIZED TOPS, AND LIGHTWEIGHT CARDIGANS, AS WELL AS TRENDY SHOES,
        ACCESSORIES, JEWELRY, ACTIVEWEAR, LOUNGEWEAR, AND SO MUCH MORE. DRESS
        UP&apos;S TRENDY, YET WEARABLE CLOTHES WILL HAVE YOU FEELING LIKE YOUR
        BEST SELF, NO MATTER WHERE YOU&apos;RE HEADED. YOU CAN SHOP WITH US
        ONLINE OR AT ANY OF OUR RETAIL LOCATIONS ACROSS THE SOUTHEAST, INCLUDING
        STORES IN GEORGIA, ALABAMA, SOUTH CAROLINA, AND TENNESSEE. GET FREE
        SHIPPING ON ALL ORDERS OVER $49, AND FEEL FREE TO REACH OUT TO US AT
        678.989.2006 WITH ANY QUESTIONS. DON&apos;T MISS OUT - COME SEE WHY
        DRESS UP IS ONE OF THE FASTEST GROWING ONLINE DRESS BOUTIQUES IN THE
        UNITED STATES!
      </p>
      <Button>SHOP OUR NEWEST ARRIVALS</Button>
    </HomeDescriptionSection>
  );
});
