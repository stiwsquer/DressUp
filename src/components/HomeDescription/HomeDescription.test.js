import HomeDescription from "./HomeDescription";
import Button from "../Button/Button";
import { BrowserRouter as Router } from "react-router-dom";

it("renders HomeDescription component without crashing", () => {
  shallow(
    <Router>
      <HomeDescription />
    </Router>
  );
});

it("contains Button component", () => {
  const wrapper = mount(
    <Router>
      <HomeDescription />
    </Router>
  );
  // console.log(wrapper.debug());
  expect(wrapper.contains(<Button>SHOP OUR NEWEST ARRIVALS</Button>)).toBe(
    true
  );
});

it("contains h2 element", () => {
  const wrapper = mount(
    <Router>
      <HomeDescription />
    </Router>
  );
  expect(wrapper.find("h2")).toHaveLength(1);
});

it("contains p element", () => {
  const wrapper = mount(
    <Router>
      <HomeDescription />
    </Router>
  );
  expect(wrapper.find("p")).toHaveLength(1);
});

it("HomeDescription snapshop - shallow", () => {
  const wrapper = shallow(
    <Router>
      <HomeDescription />
    </Router>
  );
  expect(wrapper).toMatchSnapshot();
});

it("HomeDescription snapshop - render", () => {
  const wrapper = render(
    <Router>
      <HomeDescription />
    </Router>
  );
  expect(wrapper).toMatchSnapshot();
});

it("HomeDescription snapshop - mount", () => {
  const wrapper = mount(
    <Router>
      <HomeDescription />
    </Router>
  );
  expect(wrapper).toMatchSnapshot();
});
