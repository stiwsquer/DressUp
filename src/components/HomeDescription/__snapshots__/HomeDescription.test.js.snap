// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeDescription snapshop - mount 1`] = `
<section
  className="sc-gsTEea dhRCVM"
>
  <h2>
    Trendy Clothing & Accessories at Dress Up - An Online Dress Boutique
  </h2>
  <p>
    DRESS UP IS A WOMEN'S CLOTHING BOUTIQUE CARRYING A VARIETY OF FITS, STYLES, AND PATTERNS FOR WOMEN OF ALL AGES, SHAPES, AND SIZES. WE OFFER A DIVERSE COLLECTION OF NEW PRODUCTS EVERY SINGLE WEEK, INCLUDING YOUR FAVORITE CLOTHING ITEMS: CUTE ROMPERS, MAXI DRESSES, KNIT SWEATERS, OVERSIZED TOPS, AND LIGHTWEIGHT CARDIGANS, AS WELL AS TRENDY SHOES, ACCESSORIES, JEWELRY, ACTIVEWEAR, LOUNGEWEAR, AND SO MUCH MORE. DRESS UP'S TRENDY, YET WEARABLE CLOTHES WILL HAVE YOU FEELING LIKE YOUR BEST SELF, NO MATTER WHERE YOU'RE HEADED. YOU CAN SHOP WITH US ONLINE OR AT ANY OF OUR RETAIL LOCATIONS ACROSS THE SOUTHEAST, INCLUDING STORES IN GEORGIA, ALABAMA, SOUTH CAROLINA, AND TENNESSEE. GET FREE SHIPPING ON ALL ORDERS OVER $49, AND FEEL FREE TO REACH OUT TO US AT 678.989.2006 WITH ANY QUESTIONS. DON'T MISS OUT - COME SEE WHY DRESS UP IS ONE OF THE FASTEST GROWING ONLINE DRESS BOUTIQUES IN THE UNITED STATES!
  </p>
  <button
    className="sc-bdfBQB gRvcyI"
  >
    SHOP OUR NEWEST ARRIVALS
  </button>
</section>
`;

exports[`HomeDescription snapshop - render 1`] = `
<section
  class="sc-gsTEea dhRCVM"
>
  <h2>
    Trendy Clothing & Accessories at Dress Up - An Online Dress Boutique
  </h2>
  <p>
    DRESS UP IS A WOMEN'S CLOTHING BOUTIQUE CARRYING A VARIETY OF FITS, STYLES, AND PATTERNS FOR WOMEN OF ALL AGES, SHAPES, AND SIZES. WE OFFER A DIVERSE COLLECTION OF NEW PRODUCTS EVERY SINGLE WEEK, INCLUDING YOUR FAVORITE CLOTHING ITEMS: CUTE ROMPERS, MAXI DRESSES, KNIT SWEATERS, OVERSIZED TOPS, AND LIGHTWEIGHT CARDIGANS, AS WELL AS TRENDY SHOES, ACCESSORIES, JEWELRY, ACTIVEWEAR, LOUNGEWEAR, AND SO MUCH MORE. DRESS UP'S TRENDY, YET WEARABLE CLOTHES WILL HAVE YOU FEELING LIKE YOUR BEST SELF, NO MATTER WHERE YOU'RE HEADED. YOU CAN SHOP WITH US ONLINE OR AT ANY OF OUR RETAIL LOCATIONS ACROSS THE SOUTHEAST, INCLUDING STORES IN GEORGIA, ALABAMA, SOUTH CAROLINA, AND TENNESSEE. GET FREE SHIPPING ON ALL ORDERS OVER $49, AND FEEL FREE TO REACH OUT TO US AT 678.989.2006 WITH ANY QUESTIONS. DON'T MISS OUT - COME SEE WHY DRESS UP IS ONE OF THE FASTEST GROWING ONLINE DRESS BOUTIQUES IN THE UNITED STATES!
  </p>
  <button
    class="sc-bdfBQB gRvcyI"
  >
    SHOP OUR NEWEST ARRIVALS
  </button>
</section>
`;

exports[`HomeDescription snapshop - shallow 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Memo(HomeDescription) />
</Router>
`;
